documentation/src/test/java/org/hibernate/userguide/naming/AcmeCorpPhysicalNamingStrategy.java

```diff
@@ -69,50 +65,15 @@ public Identifier toPhysicalSequenceName(Identifier name, JdbcEnvironment jdbcEn
 	public Identifier toPhysicalColumnName(Identifier name, JdbcEnvironment jdbcEnvironment) {
 		final List<String> parts = splitAndReplace( name.getText() );
 		return jdbcEnvironment.getIdentifierHelper().toIdentifier(
-				join( parts ),
+				StringUtils.join( parts, '_' ),
 				name.isQuoted()
 		);
 	}
 
-	private static Map<String, String> buildAbbreviationMap() {
-		TreeMap<String,String> abbreviationMap = new TreeMap<> ( String.CASE_INSENSITIVE_ORDER );
-		abbreviationMap.put( "account", "acct" );
-		abbreviationMap.put( "number", "num" );
-		return abbreviationMap;
-	}
-
-	private LinkedList<String> splitAndReplace(String name) {
-		LinkedList<String> result = new LinkedList<>();
-		for ( String part : StringUtils.splitByCharacterTypeCamelCase( name ) ) {
-			if ( part == null || part.trim().isEmpty() ) {
-				// skip null and space
-				continue;
-			}
-			part = applyAbbreviationReplacement( part );
-			result.add( part.toLowerCase( Locale.ROOT ) );
-		}
-		return result;
-	}
-
-	private String applyAbbreviationReplacement(String word) {
-		if ( ABBREVIATIONS.containsKey( word ) ) {
-			return ABBREVIATIONS.get( word );
-		}
-
-		return word;
-	}
-
-	private String join(List<String> parts) {
-		boolean firstPass = true;
-		String separator = "";
-		StringBuilder joined = new StringBuilder();
-		for ( String part : parts ) {
-			joined.append( separator ).append( part );
-			if ( firstPass ) {
-				firstPass = false;
-				separator = "_";
-			}
-		}
-		return joined.toString();
+	private List<String> splitAndReplace(String name) {
+		return Arrays.stream( StringUtils.splitByCharacterTypeCamelCase( name ) )
+				.filter( StringUtils::isNotBlank )
+				.map( p -> ABBREVIATIONS.getOrDefault( p, p ).toLowerCase( Locale.ROOT ) )
+				.collect( Collectors.toList() );
```

Comment:
The usage of JDK stream here is to make the code easier to understand and let user guide reader focus more on the key issue, i.e. naming strategy.